File: BackEnd/src/domain/verificationToken.ts
import { UserId, VerificationTokenId } from './ids';

export enum VerificationTokenType {
  EMAIL_VERIFICATION = 'EmailVerification',
  PASSWORD_RESET = 'PasswordReset',
}

export class VerificationToken {
  verificationTokenId: VerificationTokenId;
  userId: UserId;
  tokenHash: string;
  type: VerificationTokenType;
  createdAt: Date;
  expiresAt: Date;
  usedAt?: Date;
}

File: Frontend/GP/Models/VerificationToken.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class VerificationToken : BaseEntity
    {
        [Key]

        public int VerificationTokenId { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public string TokenHash { get; set; }
        public string Type { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime ExpiresAt { get; set; }
        public DateTime? UsedAt { get; set; }
    }
}
